monkey/infection_monkey/network_scanning/ping_scanner.py

```diff
@@ -6,16 +6,26 @@
 import sys
 
 from infection_monkey.i_puppet import PingScanData
+from infection_monkey.utils.environment import is_windows_os
 
 TTL_REGEX = re.compile(r"TTL=([0-9]+)\b", re.IGNORECASE)
 LINUX_TTL = 64  # Windows TTL is 128
 PING_EXIT_TIMEOUT = 10
+EMPTY_PING_SCAN = PingScanData(False, None)
 
 logger = logging.getLogger(__name__)
 
 
-def ping(host: str, timeout: float) -> PingScanData:
-    if "win32" == sys.platform:
+def try_ping(host: str, timeout: float) -> PingScanData:
+    try:
+        return _ping(host, timeout)
+    except Exception:
+        logging.exception("Unhandled exception occurred while running ping")
+        return EMPTY_PING_SCAN
```

Comment:
I don't think the function name `try_ping()` adds any value here. Just name it `ping()` and add a docstring that defines its behavior. The caller shouldn't need to worry bout the specific mechanisms that the function is using to handle errors.

Reply:
`try_` defines an exception wrapper. It doesn't expose anything about the mechanism, just that it's an exception handling wrapper